lib/storage/providers/SQLiteStorage.js

```diff
@@ -88,11 +88,27 @@ const provider = {
              ON CONFLICT DO UPDATE
              SET valueJSON = JSON_PATCH(valueJSON, JSON(:value));
         `;
-        const queryArguments = _.map(pairs, (pair) => {
+
+        const queryArguments = [];
+        const keysToDelete = [];
+        _.forEach(pairs, (pair) => {
+            if (pair[1] == null) {
+                keysToDelete.push(pair[0]);
+                return;
+            }
+
             const value = JSON.stringify(pair[1]);
-            return [pair[0], value];
+
+            queryArguments.push([pair[0], value]);
         });
-        return db.executeBatchAsync([[query, queryArguments]]);
+
+        const deletePromise = this.removeItems(keysToDelete);
```

Comment:
Can you add a code comment explaining why there is deleting happening inside of the merge function?